packages/react/react-types.d.ts

```diff
@@ -0,0 +1,17 @@
+import type { Size } from '@digdir/designsystemet-react';
+import type { Color } from '@digdir/designsystemet-react/colors';
+
+declare global {
+  namespace React.JSX {
+    interface IntrinsicAttributes {
+      'data-size'?: Size;
+      'data-color'?: Color;
+    }
+  }
+  namespace React {
+    interface HTMLAttributes<T> {
+      'data-size'?: Size;
+      'data-color'?: Color;
```

Comment:
These definitions crash with our components, in addition we need to support random strings incase people are using `data-color` for their own thing or some other library also uses this.

Tested by adding this type to `stories.tsconfig`
Examples that cause type errors now for me:
- Components that have other sizing scale defined (this will also be a problem for those who want to make their own scales)
<img width="783" alt="image" src="https://github.com/user-attachments/assets/58905804-4d42-4c80-bb0f-fc72f81b0265">
- All JSX elements with `data-color` are now locked to our color types. Maybe a bit to restrictive?  
<img width="1290" alt="image" src="https://github.com/user-attachments/assets/c2b7cb4d-5e1e-42a6-b260-3bdc7b18cb30">

Reply:
These are both good points, which I somehow missed when testing.

The first problem was due to the data-attributes defined in `IntrinsicAttributes` being added even when omitting the keys — I had misunderstood how `IntrinsicAttributes` worked.

For the second problem: I agree it makes sense to relax the type a bit, even though including the .d.ts file is optional.

For both, it seems best to change this file to
```
import type { Size } from '@digdir/designsystemet-react';
import type { Color } from '@digdir/designsystemet-react/colors';

declare global {
  namespace React {
    interface HTMLAttributes<T> {
      'data-size'?: Size | (string & {});
      'data-color'?: Color | (string & {});
    }
  }
}
```
which gives editor hints while still allowing arbitrary strings.

Reply:
I must question this, though:
> This will also be a problem for those who want to make their own scales

Is that something we need to support? We use `"sm"` `"md"` and `"lg"` explicitly in some components, so these will always need to exist, and our types in e.g.`Heading`, `Spinner` and `Paragraph` components reference the rest of the scale as well
```ts
export type HeadingProps {
  // ...
  'data-size'?: '2xs' | 'xs' | 'sm' | 'md' | 'lg' | 'xl' | '2xl';
}
```

So it would need to be sizes in addition to the ones we define 🤔

Reply:
> Maybe we need to split the data-size and data-color?

I would suggest just creating your own .d.ts file in that case. Just go to definition and copy the whole file, paste and change to your heart's content 😊

Reply:
> > The scaling is based of font-size now, so in theory they can just create their own classes or any CSS that defines font-size in accordance to wanted scale
> 
> Exactly, and this doesn't change that. If you make a `.size-small { fontSize = 11px }` and use that class, then it doesn't affect or is affected by `data-size` at all. It would only be an irritation if we expect people to define their own `[data-size="mySizeName"]` selectors. Those would be in addition to the ones we provide, but we wouldn't be able to provide editor hints for them. Since we now allow any string in `data-size` it's not a problem?

Aye, good point. Lets test it as is and get user feedback :)